spec/models/user_spec.rb

```diff
@@ -6,13 +6,33 @@
   describe 'validations' do
     it 'has a valid factory' do
       user = create :user
-
       expect(user).to be_valid
     end
 
     it { should validate_presence_of(:first_name) }
     it { should validate_presence_of(:last_name) }
     it { should validate_presence_of(:email) }
-    it { should validate_presence_of(:github_username) }
+    # it { should validate_presence_of(:github_username) }
```

Comment:
Do we need this validation any longer? Can we just remove it, or is it saved for the future for reasons?

Reply:
is there, `validates_uniqueness_of`?

Reply:
No, there is not.  We'd have to hand roll it.

Reply:
Looks like [shoulda_matchers](https://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord:validate_uniqueness_of) does have a `validates_uniquness_of` matcher

Reply:
Nice!

Reply:
Never used those before, is that sort of just testing that rails has that method applied? 
I guess the better way would be to create two users with the same github_username and `expect{ SecondUser.save! } to throw_error(DuplicateEntryError)`
?

Reply:
Yes testing the behavior as you described is both more thorough and appropriate imo

Reply:
@RobAWilkinson - IMO, we're good to go with Thoughtbot's shoulda matchers.  You can just do:

```ruby
it { should validate_uniqueness_of(:email) }
it { should validate_uniqueness_of(:github_username) }
```